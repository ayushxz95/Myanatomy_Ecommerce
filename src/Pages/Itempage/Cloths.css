.grid-container {
  margin-top: 50px;
  display: grid;
  gap: 50px 100px;
  padding:10px;
  grid-template-columns: auto auto auto;
    /* display: flex;
    flex-direction: row;
    justify-content: space-between;
    padding: 20px; */
}