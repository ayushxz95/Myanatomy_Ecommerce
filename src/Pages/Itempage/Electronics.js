import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchItems } from '../../Redux/Actions/ItemAction';
import Navbar from '../../Component/Navbar/Navbar';
import Card from '../../Component/Card/Card';
import './Electronics.css';

class Electronics extends Component {
      componentDidMount() {
          this.props.fetchItems();
      }
      render(){
        const {items} = this.props;
        const Electronics = items.filter(item => (item.type === 'Electronics'));
                  return ( 
                      <>
                      <div><Navbar/></div>
                      <div className='gridd-container'>
                         {  
                            Electronics.map(item => (
                              <Card key={item.id} type={item.type} name={item.name} company={item.company} id={item.key}/>
                              ))  
                         }
                       </div>
                       </>
                      );
              
          }
  }  
  const mapStateToProps = state => {
      return {
          items: state.items.vals,
          newItem: state.items.val
      }
  };
  const mapDispatchToProps = dispatch => {
      return {
      fetchItems: () => dispatch(fetchItems())
  }};
    
  export default connect(mapStateToProps, mapDispatchToProps)(Electronics);