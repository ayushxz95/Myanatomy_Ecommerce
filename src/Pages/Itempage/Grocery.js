import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchItems } from '../../Redux/Actions/ItemAction';
import Navbar from '../../Component/Navbar/Navbar';
import './Grocery.css';
import Card from '../../Component/Card/Card';

class Grocery extends Component {
    componentDidMount() {
        this.props.fetchItems();
    }
    render(){
        const {items} = this.props;
        const Grocery = items.filter(item => (item.type === 'Grocery'));
       return ( 
           <>
           <div><Navbar/></div>
           <div className='itemcontainer'>
              {
                    Grocery.map(item => (
                    <Card key={item.id} type={item.type} name={item.name} company={item.company} id={item.key}/>
                    ))
               }
            </div>
            </>
        );
     }
}  
const mapStateToProps = state => {
    return {
        items: state.items.vals,
        newItem: state.items.val
    }
};
const mapDispatchToProps = dispatch => {
    return {
    fetchItems: () => dispatch(fetchItems())
}};
  
export default connect(mapStateToProps, mapDispatchToProps)(Grocery);


