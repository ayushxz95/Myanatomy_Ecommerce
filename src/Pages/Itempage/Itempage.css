.content-container {
  display: flex;
  flex-direction: row;
}
.itemcontainer {
  /* margin-top: 20px; */
  /* display: grid; */
  /* gap: 20px 40px; */
  /* padding:10px; */
  /* display: flex;
  flex-wrap: wrap; */
  /* grid-template-columns: auto auto auto ; */
      display: grid;
      grid-gap: 12px;
      padding: 16px;
      position: relative;

  }
  @media (min-width: 600px){
    .itemcontainer{
    grid-template-columns: repeat(2,1fr);
    }
  }
  @media (min-width: 960px){
    .itemcontainer{
    grid-template-columns: repeat(3,1fr);
    }
  }
  @media (min-width: 1400px){
    .itemcontainer{
    grid-template-columns: repeat(4,1fr);
    }
  }
  @media (min-width: 1600px){
    .itemcontainer{
    grid-template-columns: repeat(5,1fr);
    }
  }
  .Add-item{
    display: inline-block;
    color: black;
    text-align: center;
    margin-top: 70px;
    right: 6px;
    position: fixed;
   
  }
  .Add-item:hover {
    /* background-color: gray; */
    cursor: pointer;
  }
  .Add-item-vacant{
    display: inline-block;
    color: black;
    text-align: center;
    top: 0;
    position: absolute;
    right: 0;
  }
  .Add-item-vacant:hover {
    /* background-color: gray; */
    cursor: pointer;
  }
  .emoticon {
    display: flex;
    font-weight: bold;
    margin-top: 70px;
    align-items: center;
    justify-content: center;
  }
  .no-item-message {
    display: flex;
    font-weight: bold;
    margin-top: 70px;
    align-items: center;
    justify-content: center;
  }
  /* .empty-state {
    display: flex;
    flex-direction: row;
  } */
  .no-item-container {
    position: relative;
  }
