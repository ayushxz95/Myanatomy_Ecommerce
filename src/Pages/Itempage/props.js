import { fetchItems } from '../../Redux/Actions/ItemAction';


const mapStateToProps = state => {
    // return {
    //     items: state.items.categoryVals,
    //     newItem: state.items.val,
    //     category: state.category.category,
    // }
    return {
        allItems: state.items.allItems,
        category: state.category.category,
        categoryItems: state.items.categoryItems,
        searchedItems: state.items.searchedItems,
    }
};
const mapDispatchToProps = dispatch => {
    return {
    fetchItems: () => dispatch(fetchItems())
}};

export {mapStateToProps, mapDispatchToProps};