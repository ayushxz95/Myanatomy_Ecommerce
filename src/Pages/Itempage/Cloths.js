import React, { Component } from 'react';
import { connect } from 'react-redux';
import { fetchItems } from '../../Redux/Actions/ItemAction';
import Navbar from '../../Component/Navbar/Navbar';
import './Cloths.css';
import Card from '../../Component/Card/Card';

class Cloths extends Component {
    componentDidMount() {
        this.props.fetchItems();
    }
    
    render(){
        const {items} = this.props;
        const Cloths = items.filter(item => (item.type === 'Cloths'));
                return ( 
                    <>
                    <div><Navbar/></div>
                    <div className='grid-container'>
                        {     
                        
                                Cloths.map(item => (
                                <Card key={item.id} type={item.type} name={item.name} company={item.company} id={item.id}/>
                                ))
                            
                        }
                        </div>
                        </>
                    );
            
        }
}  
const mapStateToProps = state => {
    return {
        items: state.items.vals,
        newItem: state.items.val
    }
};
const mapDispatchToProps = dispatch => {
    return {
    fetchItems: () => dispatch(fetchItems()),
}};
  
export default connect(mapStateToProps, mapDispatchToProps)(Cloths);


