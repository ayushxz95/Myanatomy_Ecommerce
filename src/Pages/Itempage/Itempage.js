import React, { Component } from "react";
import Navbar from "../../Component/Navbar/index";
import "./Itempage.css";
import Card from "../../Component/Card/index";
import PropTypes from "prop-types";
import { CgSmileSad } from "react-icons/cg";
import FormDialog from "../../Component/FormDialog/index";



class Itempage extends Component {
  constructor(props) {
    super();
    this.state = {
      showModal: false,
    };
  }
  componentDidMount() {
    // console.log("Hi");
    this.props.fetchItems();
  }
  openModelClickHandler = () => {
    this.setState({
      showModal: !this.state.showModal,
    });
  };
  render() {
    let items = this.props.searchedItems;
      if (Object.keys(items).length === 0) {
      return (
        <div>
          <Navbar />
          <div className="no-item-container">
          <div className="emoticon"><CgSmileSad color="ef257a" size="120px"/></div>
          <div className="no-item-message">No Item Found</div>
          <div className="Add-item-vacant">
                <FormDialog
                    message={"Add Product"}
                    buttonmessage={"Add"}
                    type={""}
                    company={""}
                    name={""}
                    id={""}
                />
          </div>
        </div>
        </div>
      );
    } else {
      return (
        <div>
          <Navbar />
          <div className="content-container">
          <div className="itemcontainer">
            {Object.keys(items).map((key) => (
              <Card
                key={items[key].id}
                type={items[key].type}
                name={items[key].name}
                company={items[key].company}
                id={items[key].id}
                image={items[key].image}
              />
            ))}
          </div>
              <div className="Add-item">
                <FormDialog
                    message={"Add Product"}
                    buttonmessage={"Add"}
                    type={""}
                    company={""}
                    name={""}
                    id={""}
                />
                </div>
          </div>
        </div>
      );
    }
  }
}

Itempage.propTypes = {
  fetchItems: PropTypes.func,
};

export default Itempage;
