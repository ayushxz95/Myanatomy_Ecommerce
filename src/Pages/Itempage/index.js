import Itempage from "./Itempage";
import {connect} from "react-redux";
import { mapDispatchToProps, mapStateToProps } from "./props";

export default connect(mapStateToProps,mapDispatchToProps)(Itempage);