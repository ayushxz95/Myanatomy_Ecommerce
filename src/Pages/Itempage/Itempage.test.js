import { shallow, configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import Itempage from "./Itempage";

configure({ adapter: new Adapter() });


describe("Testing Itempage", () => {
it("matches snapshot", () => {
    const itemObject = [{
        name: "Fan",
        company: "Bajaj",
        id: "100",
        type: "Electronics",
    }];
    const fetchItemsMock = jest.fn();
    const wrapper = shallow(<Itempage
    items={itemObject}
    fetchItems={fetchItemsMock}
    />);
    expect(wrapper).toMatchSnapshot();
});
});
//testconstant
//Use describe
//const tree = component.debug();
//expect(tree).toMatchSnapshot();