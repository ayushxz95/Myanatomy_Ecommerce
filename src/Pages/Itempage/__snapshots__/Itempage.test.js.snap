// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Itempage matches snapshot 1`] = `
<div>
  <Connect(Navbar) />
  <div
    className="content-container"
  >
    <div
      className="itemcontainer"
    >
      <Connect(Card)
        company="Bajaj"
        id="100"
        key="100"
        name="Fan"
        type="Electronics"
      />
    </div>
    <div
      className="Add-item"
    >
      <FormDialog
        buttonmessage="Add"
        company=""
        id=""
        message="Add Product"
        name=""
        type=""
      />
    </div>
  </div>
</div>
`;
